upload/upload_test.go

```diff
@@ -0,0 +1,173 @@
+host_base = inbox.test.ega.nbis.se
+encoding = UTF-8
+host_bucket = inbox.test.ega.nbis.se
+multipart_chunk_size_mb = 50
+access_key = ega-box-8902
+use_https = True
+check_ssl_certificate = False
+check_ssl_hostname = False
+socket_timeout = 30
+human_readable_sizes = True
+guess_mime_type = True
+encrypt = False
+`
+	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(configPath.Name())
+
+	err = ioutil.WriteFile(configPath.Name(), []byte(confFile), 0644)
+	if err != nil {
+		log.Fatalf("failed to write temp config file, %v", err)
+	}
+
+	_, err = loadConfigFile(configPath.Name())
+	assert.EqualError(suite.T(), err, "failed to find credentials in configuration file")
+}
+
+func (suite *TestSuite) TestConfig() {
+	var confFile = `
+access_token = eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDM0ODQxNTIsImlzcyI6Imh0dHBzOi8vbG9naW4udGVzdC5lZ2EubmJpcy5zZSIsInN1YiI6ImVnYS1ib3gtODkwMiJ9.JS-R4GUODAzbYzYs1hJR_92D9I3blfcGu5_yrK75L6QnIK6Aq6Po6kBbngJa-N9SRIzETxy5xd5uZthHaOQpFQ
+host_base = inbox.test.ega.nbis.se
+encoding = UTF-8
+host_bucket = inbox.test.ega.nbis.se
+multipart_chunk_size_mb = 50
+secret_key = ega-box-8902
+access_key = ega-box-8902
+use_https = True
+check_ssl_certificate = False
+check_ssl_hostname = False
+socket_timeout = 30
+human_readable_sizes = True
+guess_mime_type = True
+encrypt = False
+`
+	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(configPath.Name())
+
+	err = ioutil.WriteFile(configPath.Name(), []byte(confFile), 0644)
+	if err != nil {
+		log.Fatalf("failed to write temp config file, %v", err)
+	}
+
+	_, err = loadConfigFile(configPath.Name())
+	assert.NoError(suite.T(), err)
+}
+
+func (suite *TestSuite) TestSampleNoFiles() {
+
+	os.Args = []string{"upload", "-config", "upload/s3cmd.conf"}
+
+	err := Upload(os.Args)
+	assert.EqualError(suite.T(), err, "no files to upload")
+}
+
+func (suite *TestSuite) TestWrongCredentials() {
+	var confFile = `
+access_token = GciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDM0ODQxNTIHBzOi8vbG9naW4udGVzdC5lZ2EubmJpcy5zZSIsInN1YiI6ImVnYS1ib3gtODkwMiJ9.JS-R4GUODAzbYzYs1hJR_92D9I3blfcGu5_yrK75L6QnIK6Aq6Po6kBbngJa-N9SRIzETxy5xd5uZthHaOQpFQ
+host_base = inbox.test.ega.nbis.se
+encoding = UTF-8
+host_bucket = inbox.test.ega.nbis.se
+multipart_chunk_size_mb = 50
+secret_key = ega-box-8902
+access_key = ega-box-890
+use_https = True
+check_ssl_certificate = False
+check_ssl_hostname = False
+socket_timeout = 30
+human_readable_sizes = True
+guess_mime_type = True
+encrypt = False
+`
+	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(configPath.Name())
+	log.Print(configPath.Name())
+
+	err = ioutil.WriteFile(configPath.Name(), []byte(confFile), 0644)
+	if err != nil {
+		log.Fatalf("failed to write temp config file, %v", err)
+	}
```

Comment:
Since the config file is used by multiple instances put it in a struct so there are only one place to change stuff, then each test can change things to be tested before it writes it to a temp file at the beginning of the test.